Backpack/FloatingNotification/Classes/BPKFloatingNotification.swift

```diff
@@ -0,0 +1,186 @@
+        label.fontStyle = .textFootnote
+        label.textColor = BPKColor.textOnDarkColor
+        label.textAlignment = .natural
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        return label
+    }()
+    
+    private let button: BPKButton = {
+        let button = BPKButton(size: .default, style: .primary)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentHuggingPriority(.required, for: .horizontal)
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return button
+    }()
+    
+    // MARK: Initialisation
+    public init() {
+        super.init(frame: .zero)
+        button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+        
+        animator.delegate = self
+        alpha = 0
+        backgroundColor = BPKColor.corePrimaryColor
+        layer.cornerRadius = BPKCornerRadiusMd
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubviews()
+        BPKShadow.shadowLg().apply(to: layer)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func addSubviews() {
+        [iconView, messageLabel, button].forEach { stackView.addArrangedSubview($0) }
+        addSubview(stackView)
+    }
+    
+    // MARK: Animation & rendering
+    public func show(_ viewModel: FloatingNotificationViewModel) {
+        notificationsQueue.append(viewModel)
+        renderNext()
+    }
+    
+    private func renderNext() {
+        if let nextViewModel = notificationsQueue.first, !animator.isNotificationDisplayed {
+            render(viewModel: nextViewModel)
+            animator.animateUp(hideAfter: nextViewModel.hideAfter)
+        } else if animator.isNotificationDisplayed {
+            animator.animateDownNow()
+        }
+    }
+    
+    private func render(viewModel: FloatingNotificationViewModel) {
+        viewModel.parentView.addSubview(self)
+        iconView.iconName = viewModel.iconName
+        messageLabel.text = viewModel.title
+        button.title = viewModel.action?.title
+        onButtonTap = viewModel.action?.action
+        onDismissal = viewModel.didDismiss
+        
+        iconView.isHidden = viewModel.iconName == nil
+        button.isHidden = viewModel.action == nil
+        
+        setupConstraints(relativeTo: viewModel.parentView)
+        viewModel.parentView.layoutIfNeeded()
+    }
+
+    private func setupConstraints(relativeTo parent: UIView) {
+        constraints.forEach { removeConstraint($0) }
+        
+        bottomConstraint = bottomAnchor.constraint(
+            equalTo: parent.safeAreaLayoutGuide.bottomAnchor,
+            constant: Constants.startBottomConstraint
+        )
+        
+        var constraints = [
```

Comment:
Could we do something with the 'specialiist' reading layout anchors? 
What happens when we're on iPad splitview? (Any of the sizes)

Reply:
Nice one, I'll base it on the max width we accept